src/main/java/net/sf/jabref/gui/BasePanel.java

```diff
@@ -950,6 +955,40 @@ private void copyKeyAndTitle() {
         }
     }
 
+    private void copyKeyAndLink() {
+        List<BibEntry> selectedEntries = mainTable.getSelectedEntries();
+        if (!selectedEntries.isEmpty()) {
+            storeCurrentEdit();
+
+            StringBuilder sb = new StringBuilder();
+
+            long copied = 0;
+            for (BibEntry entry : selectedEntries.stream().filter(BibEntry::hasCiteKey).collect(Collectors.toList())) {
+                String key = entry.getCiteKeyOptional().get();
+                String url = entry.getField(FieldName.URL).isPresent() ? entry.getField(FieldName.URL).get() : "";
+                sb.append(String.format("<a href=\"%s\">%s</a>", url, key));
+                sb.append(OS.NEWLINE);
+                copied++;
+            }
+
+            if (copied == 0) {
+                output(Localization.lang("None of the selected entries have BibTeX keys."));
+                return;
+            }
+
+            final StringSelection ss = new StringSelection(sb.toString());
```

Comment:
I just used this code snipped from the `copyKeyAndTitle` mthod but actually I think it can be replaced by the ClipboardManager's `setClipboardContents()` method.